builtin/plugins/dposv3/dpos_test.go

```diff
@@ -1270,6 +1271,254 @@ func TestElect(t *testing.T) {
 	assert.Equal(t, newTier, validator.LocktimeTier)
 }
 
+func TestZeroRewardsCap(t *testing.T) {
+	pctx := createCtx()
+	coinAddr := pctx.CreateContract(coin.Contract)
+
+	coinContract := &coin.Coin{}
+	coinCtx := pctx.WithAddress(coinAddr)
+	coinContract.Init(contractpb.WrapPluginContext(coinCtx), &coin.InitRequest{
+		Accounts: []*coin.InitialAccount{
+			makeAccount(delegatorAddress1, 1e18),
+			makeAccount(delegatorAddress2, 2000000000000000000),
+		},
+	})
+	cycleLengthSeconds := int64(100)
+	dpos, err := deployDPOSContract(pctx, &Params{
+		ValidatorCount:      1,
+		ElectionCycleLength: cycleLengthSeconds,
+		CoinContractAddress: coinAddr.MarshalPB(),
+		MaxYearlyReward:     &types.BigUInt{Value: loom.BigUInt{big.NewInt(0)}},
```

Comment:
Deploy the contract with a non-zero `MaxYearlyReward`, then use `DPOS.SetMaxYearlyReward()` to set it to zero.